### DISCUSSION TITLE
Lora sx1278 suddenly showing"unable to set sync word" error

### DISCUSSION THREAD

**Discussion by anandspathak** [jgromes/RadioLib#785] [General]
Dear sir,
After flawlessly uninterrupted  working since last few months, suddenly it shows the error"unable to set sync word" .. Sync word is 0x14....this happened when switching off the device and switching on again..suddenly this message and program stucks there....I have changed lora new one but still it shows error.... switching on off device randomly sometimes works initialization successful and then fails again if I switch off and back on...

Can you please suggest something,
By the way it has been working non stop since last two month's..
Thanks for your library,
Regards,
Anand Pathak,
India

**Comment by jgromes** [comment]
0x14 is not an officially supported sync word for LoRa. According to the chip manufacturer, you should only use 0x12 for private networks and 0x34 for LoRaWAN. Other values can be set, but may have undocumented side effects. So the first thing I would try is to change the sync word to 0x12. If that doesn't work, you will hve to provide more information: what is the exact error code - the value? Can you reproduce the issue with a minimum example? What does the hardware look like (which Arduino, how is it wired, what is the power supply etc.).

**Reply by anandspathak** [reply]
Thank you for your reply,
I will change the sync word as you suggested,
I am out of factory...
Meanwhile I am using esp32 powered by 5v 2amp smps on  5v esp32 pin, 3.3v dropped from smps using ams1117 for lora,  lora directly  connected to esp32 pins , error code is -2 , 
The device has  analog LVDT to measure distance in mm and sends data to another lora device at 15 minute time interval...
This setup was working as is since last 2 month continuously ....I modified lvdt function only and this happened....
For your information, 
Pls consider,
Thanks,
Regards,
Anand Pathak,
India

**Reply by jgromes** [reply]
> error code is -2

In that case, what lead you to the initial conclusion that sync word is the issue? -2 is a generic "module not found error", suggesting something wrong with the hardware. Debug mode output would be useful.

**Reply by anandspathak** [reply]
Sir you are correct. ..soldering issue...now got working...actually I made pcb myself at home...
Thanks for help,
Regards,
Anand Pathak
India

**Comment by anandspathak** [comment]
Sir,
I don't want to use arduino String in receive....instead I want to go with c style strings to avoid and memory problem....can you pls tell me how can I recieve using c style strings....
Thanks
Regards

**Comment by jgromes** [comment]
It's shown in the examples:

```c++
// you can also receive data as byte array
byte byteArr[8];
int state = radio.receive(byteArr, 8);
```

If you don't know the packet length in advance, you can call `radio.getPacketLength()` to retrieve it (if the radio is configured for variable length packets, which is the default).

**Reply by anandspathak** [reply]
Oh..ok...thanks...

**Comment by anandspathak** [comment]
Sir, I am using ping pong example of sx1278. Transmitter is mounted on water tank on rooftop while receiver is below on ground floor....transmitter sends water level by lora and receiver  on ground floor receives data either starts or stops motor and then sends back "0" to transmitter for sleeping for 30 minutes....
The issue is: 
 Reciever sometimes is unable to send data and so "transmission finished" is not displayed....after this receiver can no more recieve packet and have to be power cycled to start recieve again....and again repeats same issue afterwards....
can you help...I am not well versed in arduino and from accountancy background....pls suggest what can I do,
Thanks and regards
Anand Pathak

**Reply by anandspathak** [reply]
Reset pin is connected to esp32 pin number 2, I found esp32 has itself an led connected to pin 2...i think I must change that pin to some other pin ....I do and update if resolved or not....

**Reply by anandspathak** [reply]
I changed reset pin to another pin gpio22 and issue still remains...pls suggest...thanks

**Reply by anandspathak** [reply]
Update: I found that stabilizing capacitors were missing from input and output of 5v to 3.3v ams1117 converter....I soldered them and its being 100 times data transfer, not a single issue occurred till now...
Do your think of any other reason why I have that issue???

**Reply by jgromes** [reply]
If you used an LDO without capacitors, then I'm surprised it worked at all.

> its being 100 times data transfer, not a single issue occurred till now...

I don't understand whether there is an issue now or not?

**Reply by anandspathak** [reply]
Sir, I removed capacitors from LDO and issue was back again...it was mistake and now everything working good finally...it has been 3 days no error so far....
Thanks for your suggestions,
Regards
Anand Pathak
